add arguments for buttons functions (#2266)
MiczFlor/RPi-Jukebox-RFID: components/gpio_control/README.md

* **enabled**: This needs to be `True` for the button to work.
* **Pin**: GPIO number
* **functionCall**: Primary function that you want to be called on a button press. See  [function documentation below](#doc_funcs).

However, a button has more parameters than these. In the following comprehensive list you can also find the default values which are used automatically if you leave out these settings:

> * **functionCallArgs**: Primary function arguments, defaults to `None`. Arguments are ignored if `functionCall` does not take any.

Reviewer: ```suggestion
* **functionCallArgs**: Arguments for primary function, defaults to `None`. Arguments are ignored, if `functionCall` does not take any.
```

* **hold_mode**: Specifies what shall happen if the button is held pressed for longer than `hold_time`:
  * `None` (Default): Nothing special will happen.
  * `Repeat`: The configured `functionCall` is repeated after each `hold_time` interval.
  * `Postpone`: The function will not be called before `hold_time`, i.e. the button needs to be pressed this long to activate the function
  * `SecondFunc`: Holding the button for at least `hold_time` will additionally execute the function `functionCall2` with `functionCall2Args`.
  * `SecondFuncRepeat`: Like SecondFunc, but `functionCall2` is repeated after each `hold_time` interval.
  
  In every `hold_mode` except `Postpone`, the main action `functionCall` gets executed instantly.
  
  Holding the button even longer than `hold_time` will cause no further action unless you are in the `Repeat` or `SecondFuncRepeat` mode.
  
* **hold_time**: Reference time for this buttons `hold_mode` feature in seconds. Default is `0.3`. This setting is ignored if `hold_mode` is unset or `None`
* **functionCall2**: Secondary function, default is `None`. This setting is ignored unless `hold_mode` is set to `SecondFunc` or `SecondFuncRepeat`.

> * **functionCall2Args**: Secondary function arguments, default is `None`. Arguments are ignored if `functionCall2` does not take any.

Reviewer: ```suggestion
* **functionCall2Args**: Arguments for secondary function, default is `None`. Arguments are ignored, if `functionCall2` does not take any.
```

In this example, the volume will be in-/decreased step-wise using intervals of 0.3 seconds while the respective button is held. If both buttons are pushed simultaneously, the player is muted (volume 0). In this example, Pin1 is used for increasing the volume, while Pin2 decreases it.

Function arguments can also be passed to this control with the settings **functionCall1Args**, **functionCall2Args**, **functionCallTwoButtonsArgs**

Furthermore, the following settings can be used as described for the [regular buttons](#doc_button): **pull_up_down**, **edge**, **bouncetime**, **antibouncehack**

> ### RotaryEncoder<a name="doc_rotary"></a>

Reviewer: Do we still need html anchors? I think markdown can do this now with appending `#RotaryEncoder` in a link.

Author: I wondered what they are for. So its legacy? then we can remove them :)

Reviewer: I think removing should be fine

hold_time: 5.0
hold_mode: SecondFuncRepeat
functionCall: functionCallPlayerSeekFwd
functionCallArgs: 5
functionCall2: functionCallPlayerSeekFarFwd
functionCall2Args: 30
```

In this example, a short press initiates a short jump forward by customized 5 seconds (functionCallPlayerSeekFwd) while holding the button will cause further, longer jumps. In this case it will cause a jump of customized 30 seconds forward (functionCallPlayerSeekFarFwd) every 5 seconds.
For jumping backwards, this can be done equivalently (see [function list below](#doc_funcs)).

#### Use Buttons to start playlists
To use GPIO-Pins to play music, you can emulate a card scan with the function `functionCallTriggerPlayCardId`. Supply the `cardid` as `functionCallArgs`.

> This behaves like a card scan, so you must link a folder to the id (on first press it will show up in the WebUi as new card).

Reviewer: ```suggestion
This behaves like a card scan, so you must link a folder to the id (on first press it will show up in the Web App as new card).
```

* **functionCallRecordPlayLatest**: Play latest recording
* **functionCallToggleWifi**: Toggle WIFI
* **functionCallPlayerStop**: Stop Player
* **functionCallPlayerSeekFwd**: Seek 10 seconds forward (custom seconds as optional argument)
* **functionCallPlayerSeekBack**: Seek 10 seconds backward (custom seconds as optional argument)
* **functionCallPlayerSeekFarFwd**: Seek 1 minute forward (custom seconds as optional argument)
* **functionCallPlayerSeekFarBack**: Seek 1 minute backward (custom seconds as optional argument)
* **functionCallPlayerRandomTrack**: Jumps to random track (within current playlist)
* **functionCallPlayerRandomCard**: Activate a random card
* **functionCallPlayerRandomFolder**: Play a random folder

> * **functionCallBluetoothToggle**: Switches to the audio device (e.g. speaker, bluetooth headphones), default 'toogle' (mode as optional argument)

Reviewer: ```suggestion
* **functionCallBluetoothToggle**: Switches to the audio device (e.g. speaker, bluetooth headphones), default 'toggle' (mode as optional argument)
```
